planner/utilities.py

```diff
@@ -39,13 +46,39 @@ def dump_steps(*parts, final=False):
                  })]
 
 
+def get_s3_client():
+    endpoint_url = os.environ.get("S3_ENDPOINT_URL")
+    s3_client = boto3.client('s3', endpoint_url=endpoint_url)
+    if endpoint_url:
+        try:
+            s3 = boto3.resource('s3', endpoint_url=endpoint_url)
+            s3.create_bucket(Bucket=PKGSTORE_BUCKET)
+            bucket = s3.Bucket(PKGSTORE_BUCKET)
+            bucket.Acl().put(ACL='public-read')
+        except: # noqa
+            logging.exception('Failed to create the bucket')
+            pass
+    return s3_client
+
+
 def s3_path(*parts):
     if os.environ.get('PLANNER_LOCAL'):
         path = '/'.join(str(p) for p in parts)
         return path
     else:
+        client = get_s3_client()
```

Comment:
we should probably get the client only once, and not every time this function is called.